_posts/2018-07-25-dota-macos-performance.md

```diff
@@ -0,0 +1,64 @@
+
+### Results
+
+| Library         | % of Main thread | fps - simple | fps - complex/low | fps variability/low | fps - complex/high | fps variability/high |
+| --------------- | ---------------- | ------------ | ------------- | --------------- | -------- | ---- |
+| gfx / immediate | 35%              | 90 fps       | 37.0 fps      | 5.7             | 31.0 fps | 4.5  |
+| gfx / deferred  | 15%              | 110 fps      | 44.4 fps      | 5.7             | 36.2 fps | 5.3  |
+| MoltenVK        | 21%              | 100 fps      | 41.7 fps      | 6.4             | 35.2 fps | 5.7  |
+| OpenGL          | ?                | 105-110 fps  | 45.6 fps      | 5.7             | 33.9 fps | 6.8  |
+
+The first metric shows how much time of the main thread is spent inside the portability library, compared to the total execution time (which is the actual time minus all the sleeping). It was measured with Instruments on a simple scene ("fps - simple" contains the frame rate on it). Note that the submission is done on a separate thread by Dota, so that time isn't taken into account here. Interestingly, the less time we spend on the main thread the faster our frame rate ends up being. Or, in other words, it's the race of who gets to the submission first :)
+
+The "fps - complex" and "fps variability" are from the `Source2Bench.csv` produced by timing the [dota2-pts-1971360796.dem](http://www.phoronix-test-suite.com/benchmark-files/dota2-pts-1971360796.dem.tar.bz2) demo in range 40000 .. 50000. The "/low" group refers to `-autoconfig_level 1 -low` command line, while the "/high" group is for `-autoconfig_level 3 -high`. We followed the instructions provided by [GamingOnLinux](https://www.gamingonlinux.com/articles/want-to-benchmark-dota-2-on-linux-heres-how-to-do-it.7435). To be fair, we were only testing on the simple scene until this point, so more discoveries are possible once we start profiling this complex demo.
+
+#### Experiment Setup
+
+Hardware:
+  - MacBook Pro 2016
+  - 3.3 GHz Intel Core i7 (dual-core)
+  - Intel Iris Graphics 550
+
+Software: macOS 10.14 beta
+Resolution: 1440 x 900
+
+The benchmark scripts can be seen at our [gfx-portability PR](https://github.com/gfx-rs/portability/pull/118). We were running the same scene as [Phoronix](https://www.phoronix.com/scan.php?page=news_item&px=Dota-2-Initial-Mac-Vulkan), just with a longer time frame and slightly adjusted settings.
+
+### Conclusions
+
+MoltenVK does a good job translating Vulkan to Metal. However, benchmarking shows that it has subpar framerate variability (which is not desirable for a competitive game), and can even be slower than OpenGL on low settings. This doesn't match either Phoronix or MoltenVK's numbers, although we could think of the differences in the test setup that could explain that:
```

Comment:
'although ~~we could think of~~ the differences in the test setup ~~that~~ could explain that'